framework/auth/views.py

```diff
@@ -1035,11 +1035,14 @@ def external_login_email_get():
 
     form = ResendConfirmationForm(request.form)
     session = get_session()
-    if not session.is_external_first_login:
+    if session.get('auth_user_external_first_login', False) is not True:
         raise HTTPError(http_status.HTTP_401_UNAUTHORIZED)
 
-    external_id_provider = session.data['auth_user_external_id_provider']
-    auth_user_fullname = session.data.get('auth_user_fullname')
+    external_id_provider = session.get('auth_user_external_id_provider', None)
+    auth_user_fullname = session.get('auth_user_fullname', None)
+
+    if not external_id_provider:
+        raise HTTPError(http_status.HTTP_400_BAD_REQUEST)
```

Comment:
Fix 3 (Improvement):

Return back request if `external_id_provider` is not provided.